react/components/Metrics/Controllers/TimeControllers/index.tsx

```diff
@@ -0,0 +1,146 @@
+    const startDate = this.state.startDate
+    const endDate = this.state.endDate
+    const currentTime = new Date()
+    return (startDate && endDate &&
+      startDate <= endDate &&
+      endDate <= currentTime)
+  }
+
+  public setTimeMode = (mode: any) => {
+    this.setState({ mode })
+  }
+
+  public render = () => {
+    return (
+      <div>
+        {
+          this.state.mode === 'Empty'
+            ? (
+              <Fragment>
+                <EmptyState title="Please set a time range">
+                  <p>
+                    Please set a time range to analyze your metric
+                  </p>
+                </EmptyState>
+                <div className="pa4 mh2 flex justify-end">
+                  <Button
+                    variation="primary"
+                    size="small"
+                    onClick={() => this.setTimeMode('DateRange')}
+                  >
+                    Set Time Range
+                  </Button>
+                </div>
+              </Fragment>
+            ) : (
+              <div className="flex flex-wrap mh1">
+                <div className="w-40 pa2 mr1">
+                  <RadioGroup
+                    name="timeController"
+                    options={[
+                      { value: 'DateRange', label: 'DateRange' },
+                      { value: 'Relative', label: 'Relative' },
+                    ]}
+                    value={this.state.mode}
+                    onChange={(e: EventWithValue) => this.setState({
+                      mode:
+                        e.currentTarget
+                          ? e.currentTarget.value as TimeRange
+                          : 'Empty',
+                    })}
+                  />
+                </div>
+                <div className="w-40 pa2 mr1">
+                  <TimeContext.Consumer>
+                    {({ timeControllers, setTimeControllers }) => (
+                      <Fragment>
+                        {this.state.mode === 'DateRange'
+                          ? (
+                            <DateRange
+                              locale={this.state.locale}
+                              startDate={this.state.startDate}
+                              endDate={this.state.endDate}
+                              rangeStep={this.state.rangeStep}
+                              handleRangeStep={this.setRangeStep}
+                              handleStartDate={this.setStartDate}
+                              handleEndDate={this.setEndDate}
+                            />
+                          ) : (
+                            <Relative />
+                          )
+                        }
+                        <div className="pa4 mh2">
+                          <Button
+                            variation="primary"
+                            size="small"
+                            disabled={!this.isButtonActive()}
+                            onClick={() => {
```

Comment:
Since you're using a class component, I'd move this click handler to a method in order to prevent the creation of a new function on each render.